File: src/main/java/com/customer/management/service/CustomerService.java
package com.customer.management.service;

import com.customer.management.entity.Customer;

import java.util.List;

public interface CustomerService {

    Customer addCustomer(Customer customer);

    List<Customer> getAllCustomers();

    Customer getCustomerById(Long id);

    Customer updateCustomer(Long id, Customer customer);

    void deleteCustomer(Long id);

    List<Customer> searchCustomer(String keyword);

    List<Customer> getCustomersByStatus(String status);

}

File: customer-management-frontend/src/services/customerService.js
import api from './api'

export const getCustomers=()=>{
    return api.get("/customers")
}

export const getCustomerById=(id)=>{
 return api.get(`/customers/${id}`)
}

export const addCustomer=(customer)=>{
    return api.post("/customers",customer)
}

export const updateCustomer=(id,customer)=>{
    return api.put(`/customers/${id}`, customer)
}

export const deleteCustomer=(id)=>{
    return api.delete(`/customers/${id}`)
}

export const searchCustomer=(keyword) => {
    return api.get(`/customers/search?keyword=${keyword}`)
}

export const getCustomerByStatus= (status)=>{
    return api.get(`/customers/status/${status}`)
}
